soc/realtek/ec/rts5912/reg/reg_crypto.h

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2025 Realtek, SIBG-SD7
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SOC_REALTEK_RTS5912_REG_CRYPTO_H
+#define ZEPHYR_SOC_REALTEK_RTS5912_REG_CRYPTO_H
+
+typedef struct {
+	volatile uint32_t CTRL;
+	volatile const uint32_t STS;
+	volatile const uint32_t RESERVED[2];
+	volatile uint32_t DIGEST[16];
+	volatile uint32_t FIFOIN;
+} SHA2_Type;
```

Comment:
* Don't create new typedefs.
* Use snake case naming
* Move the volatile attribute to the caller, for example `volatile struct sha2_type *sha2_regs`